scene/gui/code_edit.cpp

```diff
@@ -1762,22 +1761,38 @@ void CodeEdit::unfold_line(int p_line) {
 			break;
 		}
 		_set_line_as_hidden(i, false);
-		if (is_line_code_region_start(i - 1)) {
-			set_line_background_color(i - 1, Color(0.0, 0.0, 0.0, 0.0));
-		}
 	}
-	queue_redraw();
 }
 
 void CodeEdit::fold_all_lines() {
 	for (int i = 0; i < get_line_count(); i++) {
-		fold_line(i);
+		_fold_line(i);
 	}
 	queue_redraw();
+	emit_signal(SNAME("fold_line_updated"));
+}
+
+void CodeEdit::fold_line(int p_line) {
+	_fold_line(p_line);
+	queue_redraw();
+	emit_signal(SNAME("fold_line_updated"));
 }
 
 void CodeEdit::unfold_all_lines() {
-	_unhide_all_lines();
+	for (int i = 0; i < get_line_count(); i++) {
+		_unfold_line(i);
+	}
+	// Missing _update_scrollbars() call to be exactly the same.
+	queue_accessibility_update();
+	queue_redraw();
```

Comment:
```suggestion
```
You shouldn't have to queue the accessibility update or queue redraw for any of the line folding or unfolding, since `_set_line_as_hidden` handles it. And don't worry that it gets called multiple times, since its queued.
The `_update_scrollbars`, you probably don't have to worry about, as long as it seems like the scrollbars get updated when folding/unfolding.